smartlearner/optimizers/adagrad.py

```diff
@@ -0,0 +1,55 @@
+from collections import OrderedDict
+import theano.tensor as T
+
+from . import SGD
+from ..utils import sharedX
+
+
+class ADAGRAD(SGD):
+    """
+    Implements the ADAGRAD optimizer.
+
+    Reference
+    ---------
+    Duchi, J., Hazan, E., & Singer, Y. (2010).
+    Adaptive subgradient methods for online learning and stochastic optimization.
+    Journal of Machine Learning
```

Comment:
Yep, thanks I've already modified it but forgot to push it. I also added a lablel so we can cite it from the docstring using `[Duchi11]_`.
```
References
----------
.. [Duchi11] Duchi, J., Hazan, E., & Singer, Y., "Adaptive Subgradient Methods
             for Online Learning and Stochastic Optimization", Journal of
             Machine Learning Research, vol. 12, pp. 2121-2159, 2011.
```